packages/react-components/react-table/library/src/components/DataGridRow/useDataGridRowStyles.styles.ts

```diff
@@ -1,21 +1,71 @@
-import { mergeClasses } from '@griffel/react';
+import { makeStyles, mergeClasses } from '@griffel/react';
+import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';
 import type { DataGridRowSlots, DataGridRowState } from './DataGridRow.types';
 import type { SlotClassNames } from '@fluentui/react-utilities';
 import { useTableRowStyles_unstable } from '../TableRow/useTableRowStyles.styles';
+import { useIsInTableHeader } from '../../contexts/tableHeaderContext';
+import { useDataGridContext_unstable } from '../../contexts/dataGridContext';
+import { tableSelectionCellClassNames } from '../TableSelectionCell/useTableSelectionCellStyles.styles';
 
 export const dataGridRowClassNames: SlotClassNames<DataGridRowSlots> = {
   root: 'fui-DataGridRow',
   selectionCell: 'fui-DataGridRow__selectionCell',
 };
 
+const useStyles = makeStyles({
+  singleSelectHeader: {
+    ...createCustomFocusIndicatorStyle(
+      {
+        [`& .${tableSelectionCellClassNames.root}`]: {
+          opacity: 0,
+        },
+      },
+      { selector: 'focus-within' },
+    ),
+
+    ':hover': {
+      [`& .${tableSelectionCellClassNames.root}`]: {
+        opacity: 0,
+      },
+    },
+  },
+
+  subtleSelection: {
+    ...createCustomFocusIndicatorStyle(
+      {
+        [`& .${tableSelectionCellClassNames.root}`]: {
+          opacity: 1,
+        },
+      },
+      { selector: 'focus-within' },
+    ),
+
+    ':hover': {
+      [`& .${tableSelectionCellClassNames.root}`]: {
+        opacity: 1,
+      },
+    },
+  },
+});
+
 /**
  * Apply styling to the DataGridRow slots based on the state
  */
 export const useDataGridRowStyles_unstable = (state: DataGridRowState): DataGridRowState => {
   'use no memo';
 
+  const isHeader = useIsInTableHeader();
+  const isSingleSelect = useDataGridContext_unstable(ctx => ctx.selection.selectionMode === 'single');
+  const isSubtle = useDataGridContext_unstable(ctx => ctx.subtleSelection);
```

Comment:
IMO this should be moved to state as in other components

Reply:
If this is moved to state, the only purpose would be to increase our API surface, and pass through to this hook where it is used.

Since styles hooks are hooks, what's the consequence?